### DISCUSSION TITLE
Metro Web Roadmap

### DISCUSSION THREAD

**Discussion by erisvaldojunior** [react-native-community/discussions-and-proposals#560] [Ideas]
Hi guys,

To plan our migration from Webpack to Metro Web, I would hugely appreciate a high-level roadmap of Metro Web, with estimate quarters / months for things like:

Bundle Splitting
Asset Manifests
CSS Handling
Tree Shaking

All these things are going to be addressed on short-term?

Thanks in advance!

**Comment by EvanBacon** [comment]
Bundle splitting, tree shaking, and asset manifests aren’t specific to the web as all platforms could utilize them. These could be worked on today for native platforms and later adapted to work on web. 

Metro for web requires an error overlay, fast refresh runtime, and async bundle loading runtime. There’s also a number of configuration details required to get the majority of web packages working as expected, e.g. main file resolution per-package. 

Before any of this work can commence you’d need to have a running version of Metro for web. The only open source project I know of that’s doing this at scale is Expo Router/CLI (where web support is not currently a top-priority of the project).

From the list you shared, only CSS is critical for our web support needs so it’ll probably be worked on around Q3. Given the nature of React Native for web’s CSS-in-JS, CSS is most often used as a feature of static generation, so we’d also prioritize SSG before CSS support as well.
